hedera-node/hedera-app/src/xtest/java/token/update/TokenUpdateRemoveKeysXTest.java

```diff
@@ -0,0 +1,151 @@
+            .build();
+    protected static final String TOKEN_REMOVE_KEYS_ID = "tokenRemoveKeysId";
+    protected static final String TOKEN_CHANGE_KEYS_ID = "tokenChangeKeysId";
+    private static final String TOKEN_TREASURY = "tokenTreasury";
+    private static final String FIRST_ROYALTY_COLLECTOR = "firstRoyaltyCollector";
+    private static final int PLENTY_OF_SLOTS = 10;
+
+    private static final Key DEFAULT_KEY = Key.newBuilder()
+            .ed25519(Bytes.wrap("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa".getBytes()))
+            .build();
+
+    @Override
+    protected void doScenarioOperations() {
+        // remove freeze key
+        handleAndCommitSingleTransaction(
+                component.tokenUpdateHandler(),
+                tokenUpdate(idOfNamedToken(TOKEN_REMOVE_KEYS_ID), List.of(b -> b.freezeKey(IMMUTABILITY_SENTINEL_KEY))),
+                OK);
+        // remove kyc key
+        handleAndCommitSingleTransaction(
+                component.tokenUpdateHandler(),
+                tokenUpdate(idOfNamedToken(TOKEN_REMOVE_KEYS_ID), List.of(b -> b.kycKey(IMMUTABILITY_SENTINEL_KEY))),
+                OK);
+        // remove wipe key
+        handleAndCommitSingleTransaction(
+                component.tokenUpdateHandler(),
+                tokenUpdate(idOfNamedToken(TOKEN_REMOVE_KEYS_ID), List.of(b -> b.wipeKey(IMMUTABILITY_SENTINEL_KEY))),
+                OK);
+        // remove supply key
+        handleAndCommitSingleTransaction(
+                component.tokenUpdateHandler(),
+                tokenUpdate(idOfNamedToken(TOKEN_REMOVE_KEYS_ID), List.of(b -> b.supplyKey(IMMUTABILITY_SENTINEL_KEY))),
+                OK);
+        // remove fee schedule key
+        handleAndCommitSingleTransaction(
+                component.tokenUpdateHandler(),
+                tokenUpdate(
+                        idOfNamedToken(TOKEN_REMOVE_KEYS_ID),
+                        List.of(b -> b.feeScheduleKey(IMMUTABILITY_SENTINEL_KEY))),
+                OK);
+        // remove pause key
+        handleAndCommitSingleTransaction(
+                component.tokenUpdateHandler(),
+                tokenUpdate(idOfNamedToken(TOKEN_REMOVE_KEYS_ID), List.of(b -> b.pauseKey(IMMUTABILITY_SENTINEL_KEY))),
+                OK);
+        handleAndCommitSingleTransaction(
+                component.tokenUpdateHandler(),
+                tokenUpdate(
+                        idOfNamedToken(TOKEN_REMOVE_KEYS_ID), List.of(b -> b.metadataKey(IMMUTABILITY_SENTINEL_KEY))),
+                OK);
+        // remove admin key
+        handleAndCommitSingleTransaction(
+                component.tokenUpdateHandler(),
+                tokenUpdate(idOfNamedToken(TOKEN_REMOVE_KEYS_ID), List.of(b -> b.adminKey(IMMUTABILITY_SENTINEL_KEY))),
+                OK);
+    }
+
+    @Override
+    protected void assertExpectedNfts(@NonNull ReadableKVState<NftID, Nft> nfts) {
+        // Here is where we can e.g. assert the NFT has been transferred to the counterparty
```

Comment:
We can delete these `assertExpectedNfts()` and `assertExpectedTokenRelations()` overrides since they are still no-ops, as in the superclass implementation.